### ISSUE TITLE
Edge labels orientation

### ISSUE THREAD

**Issue by filchristou** [MakieOrg/GraphMakie.jl#61]
Currently when plotting a graph, the edge labels are having by default a direction from lower vertex to higher, thus sometimes text appears upside down:
```julia
using Graphs
gc = circular_ladder_graph(5)
graphplot(gc, elabels = repr.(edges(gc)), nlabels=repr.(vertices(gc)))
```
![image](https://user-images.githubusercontent.com/27723218/154319534-8d7bc922-9093-4538-9931-105136979819.png)

Locally I am using a wrapper to output labels in a more readable way:
![image](https://user-images.githubusercontent.com/27723218/154319892-e12df11e-92a7-4b63-8d56-92e23eff7e2e.png)

Would you be interested in a PR to make that the default behavior ?

**Comment by hexaeder** [comment]
How does your code work? Checking the positions of the nodes in pixel space and changing `elabels_opposite` and `elabels_align` accordingly?

I am open to changes to the whole `elabel` positioning business, I think there are so many options that it is pretty confusing.

**Comment by filchristou** [comment]
Yes, exactly.
First I check the angle of the position of the nodes with a function looking something like this:
```julia
function angle_between(ndpos, i, j) 
    x = ndpos[i]
    y = ndpos[j]
    θ = [ atan(d[2]/d[1]) for d in [y - x] ][] |> abs
    if y[2] >= x[2] && y[1] >= x[1]
        return θ |> rad2deg
    elseif y[2] >= x[2] && y[1] <= x[1]
        return π - θ |> rad2deg
    elseif y[2] <= x[2] && y[1] <= x[1]
        return π + θ |> rad2deg
    elseif y[2] <= x[2] && y[1] >= x[1]
        return 2π - θ |> rad2deg
    end
end
```
(maybe there is one already available ?)

And then if angle is between 90 and 270 I consider it upside down and I add it in the `elabels_opposite`
```julia
    p.elabels_opposite[] = [i for (i,e) in  enumerate(edges(G)) if  90 < angle_between(p[:node_pos].val, e) < 270]
```
After I iterate through the `elabels_opposite`, I toggle `:top`  <=> `:bottom` 

Suprisingly the update is not instant and then I have to manually trigger it by doing something trivial such as `p.elabels_opposite[] = p.elabels_opposite[]`

_(Maybe this is actually an issue on its own that code like `p.elabels_align[] = fill((:right, :bottom), length(edges(gc)))` doesn't update the graph plot.)_

I think it would be nice to have it as a default behavior, because the truth is I struggled a little bit to get it working. And I am not sure that every new user will want to invest the same time.
Of course handing in a specific `elabels_align` or `elabels_opposite` should overwrite such a default behavior.

> I am open to changes to the whole elabel positioning business, I think there are so many options that it is pretty confusing.

For the time being I wouldn't have much to suggest besides this little thingy

**Comment by hexaeder** [comment]
I'd be totally on board with a new default `elabels_opposite=Makie.automatic` instead of `Int[]`, such that the labels will always end up in certain angle range. This could be incorporated somewhere here

https://github.com/JuliaPlots/GraphMakie.jl/blob/0141a0d4d8f450f587e1ecb1bd02cfa25dc318e5/src/recipes.jl#L273-L292

I am not that sure about the right interface for the align thing. Do you think this is also necessary? It seems like the choice of the proper align is kinda arbitrary anyway because it directly depends on the node enumeration anyway... but i have not thought this through. Well it might matter for curvy edges using the new `curve_distance` parameters, they tend to bent to the left and the labels should stay on the outside....

Feel free to open an issue for the update of align and opposite... this might be a problem within the `text` recipe itself though, at a first glance it seems like they are properly passed down as observables...

**Comment by filchristou** [comment]
closed by #73
